From f718eae2f27c6a39245f1ded8d52a271caae4bf9 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 10 Jun 2022 12:13:05 +0200
Subject: [PATCH 0239/1695] drivers: stm32_rif: stub functions when
 CFG_STM32_RIF is off

This change stubs functions to allow cross-platform implementation in
drivers.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b39e7df52b264960715fa82722f69af2e9faaf7c drivers: stm32_rif: stub functions when CFG_STM32_RIF is off]
Change-Id: Id1b4467859c7e9ad84ed4a0930b678067b0b4b3f
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
---
 core/include/drivers/stm32_rif.h | 49 ++++++++++++++++++++++++++++++--
 1 file changed, 47 insertions(+), 2 deletions(-)

diff --git a/core/include/drivers/stm32_rif.h b/core/include/drivers/stm32_rif.h
index 0bcdad465..0a073614e 100644
--- a/core/include/drivers/stm32_rif.h
+++ b/core/include/drivers/stm32_rif.h
@@ -49,12 +49,15 @@
 #define SEM_EN_AND_OK(cidcfgr, wl)	(((cidcfgr) & _CIDCFGR_CFEN) &&  \
 					 ((cidcfgr) & _CIDCFGR_SEMEN) && \
 					 ((cidcfgr) & _CIDCFGR_SEMWL(wl)))
-
+#ifdef CFG_STM32_RIF
 #define SEM_MODE_INCORRECT(cidcfgr)	(!((cidcfgr) & _CIDCFGR_CFEN) ||  \
 					 !((cidcfgr) & _CIDCFGR_SEMEN) || \
 					 (!((cidcfgr) &			  \
 					    _CIDCFGR_SEMWL(RIF_CID1)) &&  \
 					  ((cidcfgr) & _CIDCFGR_SEMEN)))
+#else /* CFG_STM32_RIF */
+#define SEM_MODE_INCORRECT(cidcfgr)	(cidcfgr)
+#endif /* CFG_STM32_RIF */
 
 /*
  * Structure containing RIF configuration data
@@ -89,10 +92,21 @@ struct rif_conf_data {
  *  is enabled and is taken, and Cortex A35 CID is
  *  white-listed and semaphore is already taken by Cortex A35
  */
+#ifdef CFG_STM32_RIF
 TEE_Result stm32_rif_check_access(uint32_t cidcfgr,
 				  uint32_t semcr,
 				  unsigned int nb_cid_supp,
 				  unsigned int cid_to_check);
+#else
+static inline TEE_Result
+		stm32_rif_check_access(uint32_t cidcfgr __unused,
+				       uint32_t semcr __unused,
+				       unsigned int nb_cid_supp __unused,
+				       unsigned int cid_to_check __unused)
+{
+	return TEE_SUCCESS;
+}
+#endif
 
 /*
  * Parse RIF config from Device Tree extracted information
@@ -104,16 +118,47 @@ TEE_Result stm32_rif_check_access(uint32_t cidcfgr,
  * @nb_cid_supp: Number of supported CID for the IP
  * @nb_channel: Number of channels for the IP
  */
+#ifdef CFG_STM32_RIF
 void stm32_rif_parse_cfg(uint32_t rif_conf,
 			 struct rif_conf_data *conf_data,
 			 unsigned int nb_cid_supp,
 			 unsigned int nb_channel);
-
+#else
+static inline void
+		stm32_rif_parse_cfg(uint32_t rif_conf __unused,
+				    struct rif_conf_data *conf_data __unused,
+				    unsigned int nb_cid_supp __unused,
+				    unsigned int nb_channel __unused)
+{
+}
+#endif
+
+#ifdef CFG_STM32_RIF
 /* RIF semaphore functions */
 bool stm32_rif_is_semaphore_available(vaddr_t addr);
 TEE_Result stm32_rif_acquire_semaphore(vaddr_t addr,
 				       unsigned int nb_cid_supp);
 TEE_Result stm32_rif_release_semaphore(vaddr_t addr,
 				       unsigned int nb_cid_supp);
+#else
+static inline bool stm32_rif_is_semaphore_available(vaddr_t addr __unused)
+{
+	return true;
+}
+
+static inline TEE_Result
+		stm32_rif_acquire_semaphore(vaddr_t addr __unused,
+					    unsigned int nb_cid_supp __unused)
+{
+	return TEE_SUCCESS;
+}
+
+static inline TEE_Result
+		stm32_rif_release_semaphore(vaddr_t addr __unused,
+					    unsigned int nb_cid_supp __unused)
+{
+	return TEE_SUCCESS;
+}
+#endif
 
 #endif /* __DRIVERS_STM32_RIF_H */
-- 
2.47.2

